sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/PublishEventsWithAzureIdentity.java

```
@@ -0,0 +1,89 @@
+        // For the purpose of this sample, create a service principal and set the following environment variables.
+        // See https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal for
+        // information on how to create a service principal.
+        System.setProperty("AZURE_CLIENT_ID", "<<insert-service-principal-client-id>>");
+        System.setProperty("AZURE_CLIENT_ID", "<<insert-service-principal-client-application-secret>>");
+        System.setProperty("AZURE_TENANT_ID", "<<insert-service-principal-tenant-id>>");
+
+        // DefaultAzureCredentialBuilder exists inside the azure-identity package.
+        TokenCredential credential = new DefaultAzureCredentialBuilder()
+            .build();
+
+        // Create a producer.
+        // "<<fully-qualified-namespace>>" will look similar to "{your-namespace}.servicebus.windows.net"
+        // "<<event-hub-name>>" will be the name of the Event Hub instance you created inside the Event Hubs namespace.
+        EventHubProducerAsyncClient producer = new EventHubClientBuilder()
+            .credential(
+                "<<fully-qualified-namespace>>",
+                "<<event-hub-name>>",
+                credential)
+            .buildAsyncProducerClient();
+
+        final AtomicReference<EventDataBatch> currentBatch = new AtomicReference<>(
+            producer.createBatch().block());
+
+        // The sample Flux contains three events, but it could be an infinite stream of telemetry events.
+        telemetryEvents.subscribe(event -> {
+            final EventDataBatch batch = currentBatch.get();
+            if (!batch.tryAdd(event)) {
+                producer.createBatch().map(newBatch -> {
+                    currentBatch.set(newBatch);
+
+                    // Adding that event we couldn't add before.
+                    newBatch.tryAdd(event);
```

What happens if this `tryAdd` returns `false` i.e. if an event is too big for any batch? We have no logging or error to report it back to the user. We silently drop the event.

---

Yeah. I should propagate an error here.